src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerAncestorFingerprintFacet.java

```diff
@@ -0,0 +1,32 @@
+package org.jenkinsci.plugins.docker.commons.fingerprint;
+
+import hudson.model.Fingerprint;
+import java.util.Collections;
+import java.util.Set;
+import java.util.TreeSet;
+import javax.annotation.Nonnull;
+
+/**
+ * Facet representing the fact that some Docker image was derived from this one.
+ * This facet would be added to the fingerprint corresponding to the {@code FROM} instruction in a {@code Dockerfile}.
+ * {@link #getDescendantImageId} indicates the image that was built from that file.
+ * @see DockerFingerprints#addFromFacet
+ * @see DockerDescendantFingerprintFacet
+ */
+public class DockerAncestorFingerprintFacet extends DockerRunPtrFingerprintFacet {
```

Comment:
Whilst it may be in this code - if you actually have an arbitrary `Fingerprint` that fingerprint may have both `DockerAncestorFingerprintFacet` and a `DockerDescendantFingerprintFacet`.  At this point I guess the naming is a littl confused as the object of the noun isn't there.  I guess its a `DockerDecendantOfFingerPrintFacet` and a `DockerAncenstorOfFingerprintFacet` - rather than my thinking that the Fingerprint has an ancestor facet that would point to the ancestor.

I'm willing to be swayed if others also shoot me down :smile: